restcomm/restcomm.http/src/main/java/org/mobicents/servlet/restcomm/http/AccountsEndpoint.java

```diff
@@ -252,6 +283,35 @@ protected Response putAccount(final MultivaluedMap<String, String> data, final M
         }
     }
 
+    private Client createClientFrom(final Sid accountSid, final MultivaluedMap<String, String> data) {
+        final Client.Builder builder = Client.builder();
+        final Sid sid = Sid.generate(Sid.Type.CLIENT);
+        //TODO: need to encrypt this password because it's same with Account password.
+        // Don't implement now. Opened another issue for it.
+        //String password = new Md5Hash(data.getFirst("Password")).toString();
+        String password = data.getFirst("Password");
+
+        builder.setSid(sid);
+        builder.setApiVersion(getApiVersion(data));
+        builder.setFriendlyName(data.getFirst("FriendlyName"));
+        builder.setAccountSid(accountSid);
+        builder.setLogin(data.getFirst("Login"));
+        builder.setPassword(password);
+        builder.setStatus(Client.ENABLED);
+        builder.setVoiceUrl(getUrl("VoiceUrl", data));
+        builder.setVoiceMethod(getMethod("VoiceMethod", data));
+        builder.setVoiceFallbackUrl(getUrl("VoiceFallbackUrl", data));
+        builder.setVoiceFallbackMethod(getMethod("VoiceFallbackMethod", data));
+        builder.setVoiceApplicationSid(getSid("VoiceApplicationSid", data));
+        String rootUri = configuration.getString("root-uri");
+        rootUri = StringUtils.addSuffixIfNotPresent(rootUri, "/");
+        final StringBuilder buffer = new StringBuilder();
+        buffer.append(rootUri).append(getApiVersion(data)).append("/Accounts/").append(accountSid.toString())
+                .append("/Clients/").append(sid.toString());
+        builder.setUri(URI.create(buffer.toString()));
+        return builder.build();
+    }
+
     private Account update(final Account account, final MultivaluedMap<String, String> data) {
```

Comment:
When an account updates, we should check if sip client should update also, for example on the password change